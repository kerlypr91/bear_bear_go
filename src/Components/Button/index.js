import React from 'react'
import { Button as SemanticButton } from 'semantic-ui-react'

export default function Button (props) {
  const { onClick, label } = props

  const handleClick = (event) => {
    event.preventDefault()

    if (!onClick) {
      return
    }

    onClick()
  }

  return (
    <SemanticButton
      inverted
      onClick={handleClick}
    >
      {label}
    </SemanticButton>
  )
}