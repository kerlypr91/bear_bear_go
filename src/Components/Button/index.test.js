import React from 'react';
import renderer from 'react-test-renderer';
import Button from './index';

test('Button match snapshot', () => {
  const clickSpy = jest.fn()
  const preventSpy = jest.fn()

  const component = renderer.create(
    <Button label="test" onClick={clickSpy} />
  )

  // base snapshot

  let tree = component.toJSON()

  expect(tree).toMatchSnapshot()

  // onClick prop gets called when button is cliked

  tree.props.onClick({ preventDefault: preventSpy })

  expect(preventSpy).toHaveBeenCalled()
  expect(clickSpy).toHaveBeenCalled()
});