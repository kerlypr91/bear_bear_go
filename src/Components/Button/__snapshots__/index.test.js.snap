// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button match snapshot 1`] = `
<button
  className="ui inverted button"
  onClick={[Function]}
>
  test
</button>
`;
